modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/app/TessellationIOApp.scala

```diff
@@ -45,17 +46,18 @@ import eu.timepit.refined.pureconfig._
 import fs2.Stream
 import fs2.concurrent.SignallingRef
 import org.typelevel.log4cats.slf4j.Slf4jLogger
-import pureconfig.ConfigSource
 import pureconfig.generic.auto._
 import pureconfig.module.catseffect.syntax._
 import pureconfig.module.enumeratum._
+import pureconfig.{ConfigReader, ConfigSource}
 
 abstract class TessellationIOApp[A <: CliMethod](
   name: String,
   header: String,
   clusterId: ClusterId,
   helpFlag: Boolean = true,
-  version: TessellationVersion = TessellationVersion.unsafeFrom("0.0.1")
+  version: TessellationVersion = TessellationVersion.unsafeFrom("0.0.1"),
+  val configFiles: List[String]
```

Comment:
I have a slight preference for
```scala
extends CommandIOApp(
      name,
      header,
      helpFlag,
      version.version.value
    ) {

  def configFiles: List[String]
```
and 
```scala
object Main
    extends TessellationIOApp[Run](
      name = "dag-l0",
      header = "Tessellation Node",
      version = TessellationVersion.unsafeFrom(BuildInfo.version),
      clusterId = ClusterId("6d7f1d6a-213a-4148-9d45-d7200f555ecf")
    ) {

  val configFiles = List("dag-l0.conf")
```
but you have to keep in mind initialization order and  probably you can say the same of all the other parameters of TessellationIOApp, so is fine:)

Reply:
@scasplte2 it's totally up to you